network/openziti/pkg.yaml

```diff
@@ -0,0 +1,65 @@
+name: openziti
+variant: alpine
+shell: /toolchain/bin/bash
+install:
+  - unzip
+  - zip
+  - ninja-build
+  - ninja
+  - zlib-static
+dependencies:
+  - image: "ghcr.io/siderolabs/tools:v1.7.0-4-gc844dc3"
```

Comment:
this looks a bit strange... we usually build from `base` layer which contains toolchain (tools) and don't use alpine for the build. is there anything missing?

Reply:
There is no zlib-static, zip, unzip when using scratch and its impossible to use install directive as `apk` manager doesnt exist in `scratch' variant. 
Vcpkg relies on having zip, unzip as it downloads dependencies in that format. 
Zlib-static is required to link an executable against it as there is no zlib on the host /usr/lib folder

Reply:
zlib should be available though, zip and unzip, there will be no `apk` of course.

`vcpkg` downloads dependencies? does it pin them in a secure way?

Reply:
I guess I could just change to using 'base' stage instead of explicitly binding to a tools image here. 
Build seems to work fine with that. But idk what to do with those zip, unzip, zlib-static, ninja deps

Reply:
> zlib should be available though, zip and unzip, there will be no `apk` of course.
> 
> `vcpkg` downloads dependencies? does it pin them in a secure way?

Yes it does by verifying their sha512sum as I skimmed the code.

Reply:
> I guess I could just change to using 'base' stage instead of explicitly binding to a tools image here. Build seems to work fine with that. But idk what to do with those zip, unzip, zlib-static, ninja deps


Tried to build it with `- stage: base` and `variant: alpine`.
```
5.804 /toolchain/lib/gcc/x86_64-linux-musl/13.3.0/../../../../x86_64-linux-musl/bin/ld: attempted static link of dynamic object `/lib/libz.so'
5.804 collect2: error: ld returned 1 exit status
```
Which is obviously happening cuz of `-static`.

Reply:
seems there are static binaries available from the sdk releases, let's use them rather than adding all this build dependencies here, since it's going to be a lot of work to support it

Reply:
> seems there are static binaries available from the sdk releases, let's use them rather than adding all this build dependencies here, since it's going to be a lot of work to support it

There are no static builds. 
```
$ file ziti-edge-tunnel (v1.1.1)
ziti-edge-tunnel: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=36243be4064e33f3c82f7a4e9767ffc3bed1e656, for GNU/Linux 3.2.0, with debug_info, not stripped
```

Reply:
> seems there are static binaries available from the sdk releases, let's use them rather than adding all this build dependencies here, since it's going to be a lot of work to support it

Right, so what do you think next steps would be to get this merged upstream?

Reply:
hmm, then this extension needs to add it's build time dependencies as like qemu-guest-agent for example

Reply:
> hmm, then this extension needs to add it's build time dependencies as like qemu-guest-agent for example

I guess would be hard to do as it heavily depends on vcpkg as the dependency manager. 
Maybe I could try working on creating static builds and then pulling them within the build process. Would this be okay?

Reply:
I mean pulling them from the releases page, as you suggested above.

Reply:
yes, let's try static builds from the releases page. how big are those?

Reply:
> yes, let's try static builds from the releases page. how big are those?

Well, around 5MB, I am currently working on buildin those using gh workflows, just having a linker issue with CMAKE trying to link libpthread.so instead of libpthread.a.
https://github.com/openziti/ziti-tunnel-sdk-c/issues/938